### ISSUE TITLE
Bad configuration causes misleading error message

### ISSUE THREAD

**Issue by TBugReporter** [Rookiestyle/ColoredPassword#17]
As an experiment, I tried deleting the `ColoredPassword.ForeColor*` `<Value>`s from `KeePass.config.xml`, but I kept the `<Item>`s (without any `<Value>`s) because I didn't want them moving from their current positions in the file.  To my surprise, when I started KeePass like this, it said, "The following plugin is incompatible with the current KeePass version".  I know this is an error of my own making, but I thought I'd mention it anyway, just in case you might be interested in adding error correcting code for this in your plugin (assuming that's possible).

**Comment by Rookiestyle** [comment]
What is “kept the items without values“? I don't understand how the config file looked like.

**Comment by TBugReporter** [comment]
Was:
```
		<Item>
			<Key>ColoredPassword.ForeColorDefault</Key>
			<Value>-16728064</Value>
		</Item>
		<Item>
			<Key>ColoredPassword.ForeColorDigit</Key>
			<Value>Blue</Value>
		</Item>
		<Item>
			<Key>ColoredPassword.ForeColorLower</Key>
			<Value>WindowText</Value>
		</Item>
		<Item>
			<Key>ColoredPassword.ForeColorSpecial</Key>
			<Value>Red</Value>
		</Item>
```

I changed it to:
```
		<Item>
			<Key>ColoredPassword.ForeColorDefault</Key>
		</Item>
		<Item>
			<Key>ColoredPassword.ForeColorDigit</Key>
		</Item>
		<Item>
			<Key>ColoredPassword.ForeColorLower</Key>
		</Item>
		<Item>
			<Key>ColoredPassword.ForeColorSpecial</Key>
		</Item>
```

However, on thinking about it, I don't know that the exact changes are relevant, or at least, not as relevant as the fact that KeePass reports a misconfiguration as a problem with the plugin itself rather than with the configuration.

**Comment by Rookiestyle** [comment]
*tldr*
CustomConfig expects each item to hold exactly one `key` and exactly one `value` item.
Your changes result in the config file being invalid.
Simply don't do changes like that :)


**long version / what happens in this case?**
* KeePass is quite tolerant and accepts the invalid config file. Instead of complaining, it simply considers the `key`'s value to be [null](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/null)
* ColoredPassword expects either *something* to be defined as `value` or no `key` to be defined at all - in the latter case a default value is used. This is achieved by using a KeePass built-in method called `GetString(<key from config file>, <default value if key does not exist>)`
* [Color.FromName](https://docs.microsoft.com/en-us/dotnet/api/system.drawing.color.fromname?view=netframework-3.5) is used by ColoredPassword to convert the value string into a valid color (RGB - red, green, blue + alpha))
* Color.FromName throws an exception in case *null* is provided

I don't plan to release an update with additional extra logic to manually handle changes like and to cater for different tolerance levels of KeePass and .NET.
If both `key` and `value` are deleted, it will work.

Nevertheless, I'll include error handling for that in case I release a new version.

**Comment by TBugReporter** [comment]
My thoughts on reading this:

Patient:  "Doc, it hurts when I do *this*!"
Doctor:  "Well, then, *don't do that*!

:)

(Disclaimer:  The preceding should not be misconstrued or misinterpreted as any kind of complaint.  Your response was exactly what I expected.)
